### PR TITLE
added the get users task with test for bad endpoint catch all and ret…

### PR DESCRIPTION
**Pull request by Fandasuba**
…urning the array of users.

### CODE REVIEW

**Comment by Fandasuba** [comment]
Also deleted a duped app. error handler since I forgot to delete it a few tasks earlier when stacking error codes in the same app.use.  No repeat code and all that.

File: __tests__/app.test.js [hunk 1]
```diff
@@ -301,3 +301,30 @@ describe("DELETE: deleting comments by ID", () => {
...
+            username: expect.any(String),
+            name: expect.any(String),
+            avatar_url: expect.any(String),
+          });
+        });
+      });
+  });
+  test("404: test that error catches invalid user request endpoint", () => {
+    return request(app)
+      .get("/api/userz")
```
Comment (nc-john-hutcheon) [inline __tests__/app.test.js:324]: There's no requirement to test for this on this endpoint. As this is an error that could happen at any point (e.g. if someone wrote /api/articlezz, or /api/topiczzz) it's just a general server error for incorrect endpoints. It should therefore live in it's own describe block for general 404s 👍

**Comment by nc-john-hutcheon** [comment]
Nice work Craig, just see my comment below!

**Review by nc-john-hutcheon — COMMENTED** [review]
nice work Craig, just see my comment below!

**Pull request merged**